Client/src/Components/Graph/GraphControls/AxesControl.tsx

```diff
@@ -0,0 +1,254 @@
+    const missingDatasets = []
+    datasets.forEach(dataset => {
+      let exists = false
+      dataset.data.variables.forEach(variableName => {
+        if (variableName.name == type) {
+          exists = true
+        }
+      })
+      if (exists == false) {
+        missingDatasets.push(dataset.dataset_name)
+      }
+    })
+    setXVariableMissing(missingDatasets)
+  }
+
+  const checkYVariablesExist = (type: string, datasets: IDatasetModel[]) => {
+    const missingDatasets = []
+    console.log(type)
+    datasets.forEach(dataset => {
+      let exists = false
+      dataset.data.variables.forEach(variableName => {
+        if (variableName.name == type) {
+          exists = true
+        }
+      })
+      if (exists == false) {
+        missingDatasets.push(dataset.dataset_name)
+      }
+    })
+    console.log(missingDatasets)
+    setYVariableMissing(missingDatasets)
+  }
+
+  return (
+    <>
+      <Grid container>
+        <Grid item xs={12}>
+          <Button id='settingsToggle' variant="contained" onClick={handleSettingsClick} color="primary">
+            Settings
+              {showSettings ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
+          </Button>
+        </Grid>
+        {showSettings &&
+          <>
+            <Grid item xs={12}>
+              <Box className={classes.defaultBorder}>
+                <Grid container spacing={4}>
+                  <Grid item xs={4}>
+                    <FormControl>
+                      <InputLabel id="xVariable">X Variable</InputLabel>
+                      <Select
+                        labelId="xVariable"
+                        id="xVariable"
+                        value={axes[0].variableName}
+                        autoWidth={true}
+                        onChange={handleXVariableChange}
+                      >
+                        {variables.map(variable => (
+                          <MenuItem value={variable.name}>{variable.name}</MenuItem>
+                        ))}
+                      </Select>
+                    </FormControl>
+                  </Grid>
+                  <Grid item xs={4}>
+                    <FormControl>
+                      <InputLabel id="xUnits">X Units</InputLabel>
+                      <Select
+                        labelId="xUnits"
+                        id="xUnits"
+                        value={axes[0].units}
+                        autoWidth={true}
+                        onChange={handleXUnitChange}
+                      >
+                        <MenuItem value={''}>{''}</MenuItem>
+                        {xUnits.map(type => (
+                          <MenuItem value={type}>{type}</MenuItem>
+                        ))}
+                      </Select>
+                    </FormControl>
+                  </Grid>
+                  <Grid item xs={4}>
+                    <Typography align="center">
+                      Datasets Missing X: {xVariableMissing.toString()}
```

Comment:
Hopefully in the future this can be changed to replace "X" with the actual variable name

Reply:
put it back

Reply:
thanks